### ISSUE TITLE
Adding two Criterions?

### ISSUE THREAD

**Issue by yongfei25** [twitter/torch-autograd#72]
Hi,

What are the possible ways to return the sum of two loss? `ParallelCriterion` doesn't seem to have the `:add()` function. 

Tried the code below, no luck too. But it did successfully sum the values.

```lua
local lossf1 = d.nn.ClassNLLCriterion()
local lossf2 = d.nn.ClassNLLCriterion()
local loss1 = lossf1(attentionGates, attentionY)
local loss2 = lossf2(predictions, y)
    
loss = loss1 + loss2

print(loss1.value)
print(loss2.value)
print(loss.value)

return loss

-- output
0.68027830123901	
12.994906425476	
13.675184726715	

/torch/install/share/lua/5.1/torch/Tensor.lua:462: Wrong size for view. Input size: 2x1. Output size: 1x400004
stack traceback:
	[C]: in function 'error'
	/home/yongfei/torch/install/share/lua/5.1/torch/Tensor.lua:462: in function 'gf'
	...all/share/lua/5.1/autograd/runtime/direct/DirectTape.lua:58: in function 'gradOnly'

```

**Comment by yongfei25** [comment]
Turns out it was because I had two statement sharing one `d.nn.LogSoftMax` in one of the functions. Closing :)
